lib/libzfs/libzfs_iter.c

```diff
@@ -141,7 +141,7 @@ zfs_iter_filesystems(zfs_handle_t *zhp, zfs_iter_f func, void *data)
  */
 int
 zfs_iter_snapshots(zfs_handle_t *zhp, boolean_t simple, zfs_iter_f func,
-    void *data)
+    void *data, nvlist_t *range_nvl)
```

Comment:
I think that this should take the min and max as uint64_t arguments, rather than making the caller construct the range_avl.  This minimizes the code that needs to know about how the arguments are marshalled to the kernel.